### DISCUSSION TITLE
Labeling issues

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [community/community#26050] [Actions]
<p>I have a <a href="https://github.com/NETMouse-projects/Sandbox/issues" rel="nofollow noopener">test repo</a>. and I want do the following with my issues: when I remove some <strong>X</strong> issue from any milestone label <strong>[NEEDS MILESTONE]</strong> adds. The problem is I don’t know how to write <strong>.yml</strong> file to it. I’m  beginner in GitHub actions.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/alvin-seville">@alvin-seville</a> ,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>You can use ‘issues’ with type ‘demilestoned’ to trigger the workflow(please use in ‘master’ branch), and then use rest api to add a ‘Needs Milestone’ label for the issue. Please check below docs for more detail.</p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fhelp.github.com%2Fen%2Factions%2Fautomating-your-workflow-with-github-actions%2Fevents-that-trigger-workflows%23issues-event-issues&amp;data=02%7C01%7Cv-wadzho%40microsoft.com%7Cb63e0e72a7d6454cd8c308d7989e8de9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637145679060528525&amp;sdata=6SuQT343cMkscTU7UNSlkMPLnHjo7GQ%2F2KCbRn37E9g%3D&amp;reserved=0" rel="nofollow noopener">https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#issues-event-issues</a></p>
<p><a href="https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdeveloper.github.com%2Fv3%2Fissues%2Flabels%2F%23add-labels-to-an-issue&amp;data=02%7C01%7Cv-wadzho%40microsoft.com%7Cb63e0e72a7d6454cd8c308d7989e8de9%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637145679060528525&amp;sdata=ucJE752Lpwh7tzi8JS7cQ8R6rBFV2A37Om1nFirvrWw%3D&amp;reserved=0" rel="nofollow noopener">https://developer.github.com/v3/issues/labels/#add-labels-to-an-issue</a></p>
<p>Code sample as below:</p>
<pre><code>name: milestone
on:
  issues:
    types: demilestoned
jobs:
  demilestoned:
    runs-on: [ubuntu-latest]
    steps:
      - name: get issue
        run: |
          echo ${{ github.event.issue.url }}
      - name: add label
        run: |
          curl -v -u admin:${{ secrets.GITHUB_TOKEN }} -H "Accept: application/vnd.github.antiope-preview+json" -d '{"labels": ["Needs Milestone"]}' ${{ github.event.issue.url }}/labels
</code></pre>
<p><img src="https://user-images.githubusercontent.com/52772761/181094989-66c818ad-e63c-4484-9160-d4124987393c.png" alt="Labelissue.png" data-base62-sha1="xLjRI2er3IK86ZhprJ1Uw8w8wJ" width="284" height="211"></p>
